stores/metadata.go

```diff
@@ -2931,22 +2997,25 @@ func validateSort(sortBy, sortDir string) error {
 
 // createOrUpdateSector creates a sector or updates it if it exists already. The
 // resulting ID is set on the input sector.
-// NOTE: don't rely on any other fields of the returned sector than the ID
-func createOrUpdateSector(tx *gorm.DB, sector *dbSector) error {
+func createOrUpdateSector(tx *gorm.DB, sectors []dbSector) error {
+	if len(sectors) == 0 {
+		return nil // nothing to do
+	}
 	err := tx.
-		Where(dbSector{Root: sector.Root[:]}).
 		Clauses(clause.OnConflict{
 			UpdateAll: true,
 			Columns:   []clause.Column{{Name: "root"}},
 		}).
-		Create(&sector).
+		CreateInBatches(&sectors, sectorInsertionBatchSize).
 		Error
 	if err != nil {
 		return err
-	} else if sector.ID == 0 {
-		// if it already exists, fetch it - this fallback is needed for MySQL
-		// since it doesn't support returning the ID on conflict
-		if err := tx.Where(dbSector{Root: sector.Root[:]}).Take(&sector).Error; err != nil {
+	}
+	// fetch the upserted sectors
+	for i, sector := range sectors {
+		if err := tx.Raw("SELECT * FROM sectors WHERE root = ?", sector.Root[:]).
```

Comment:
hm isn't this something we could improve upon? 500 SELECTs per batch is not ideal

Reply:
Changed it to `FindInBatches`. I thought about something like that at first but after the select didn't show up when I tested (on my node where they usually show up with 200ms+ per select) and I saw that it doesn't even show up for a 10GB file I uploaded, I forgot about it since it didn't even show up in the logs. Maybe because new insertions within the same transaction are still in memory.